datafusion/physical-plan/src/joins/sort_merge_join.rs

```diff
@@ -1512,7 +1517,10 @@ impl SMJStream {
                     };
 
                     // Push the filtered batch which contains rows passing join filter to the output
-                    if matches!(self.join_type, JoinType::Left | JoinType::LeftSemi) {
+                    if matches!(
+                        self.join_type,
+                        JoinType::Left | JoinType::LeftSemi | JoinType::Right
```

Comment:
Do we also have to handle https://docs.rs/datafusion/latest/datafusion/common/enum.JoinType.html#variant.RightSemi 🤔

Reply:
At the end of the day all types of filtered joins will be moved.
I'm planning to cover this week RightSemi, LeftAnti, RightAnti and Full, lets see how it goes